### PR TITLE
Session class

### LINKED ISSUE #166: Initial `ert_storage.client`
We will need a client to avoid duplicating code. In the first iteration, we should just expose an already-configured `requests.Session` object.

Usage:
```py
from ert_storage.client import Session

with Session() as session:
    session.get("/experiments")
```

`Session` uses a `ERT_STORAGE_CONNECTION_STRING` environment variable, much like Azure client does, to determine where to connect and which token to use. This environment variable is set by ERT for each forward-model job.

Here's how we create a pre-configured `Session` object in ERT: https://github.com/equinor/ert/blob/e0d5046b4135c1fbbcee27ee40c64eacc44289fc/ert_shared/services/storage_service.py#L70-L78

The `client` subpackage can be its own PyPI package if we care that much, just like `azure-blob-storage` is a separate package, but in the same repository as all the other Azure Python bindings.

### PR DESCRIPTION
**Pull request by TerryHannant**
Initial implementation 

Resolves #166

### CODE REVIEW

**Comment by ertomatic** [comment]
Can one of the admins verify this patch?

File: src/ert_storage/client/session.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        exc_type: Type[BaseException],
+        traceback: TracebackType,
+    ) -> bool:
+        pass
+
+    def _resolve_url(self) -> str:
+        """Resolve which of the candidate base urls to use."""
+        for url in self._connection_info["urls"]:
+            try:
+                # Original code has auth token passed but is it actually used?
```
Comment (frode-aarstad) [inline src/ert_storage/client/session.py:76]: It has been used but. I think its removed for lazy developers I would assume. Will probably be used in the future

**Comment by frode-aarstad** [comment]
Looks good to me. Perhaps @zohar should have a quick look since he is the ert-storage guru

File: src/ert_storage/client/session.py (hunk 1 above)
Comment (TerryHannant) [reply src/ert_storage/client/session.py:76]: Should the auth be added then, even though its not actually needed?

Comment (dotfloat) [reply src/ert_storage/client/session.py:76]: It's likely not needed anymore. The auth is more fine-grained now than how it was when this loop was written.

File: tests/integration/test_sessions.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import pytest
+import json
+from ert_storage.client.session import Session
+
+HTTP_OK_STATUS = 200
```
Comment (dotfloat) [inline tests/integration/test_sessions.py:5]: Nitpick: use `from fastapi import status` which provides these same constants, but in a centralised location.

**Comment by dotfloat** [comment]
Looks good to me otherwise. Remember to squash your commits before merging.

File: tests/integration/test_sessions.py (hunk 2 above)
Comment (TerryHannant) [reply tests/integration/test_sessions.py:5]: Did consider that but I have a general aversion to adding dependency to tests just for consts, Happy to change it over though

**Review by frode-aarstad — APPROVED** [review]

**Pull request merged**